tests/test_packages/test_connections/test_p2p_libp2p/test_public_dht.py

```diff
@@ -85,171 +82,74 @@ def delegate_uris_public_keys(request):
-            multiplexer.connect()
-            assert connection.is_connected, f"Couldn't connect to {uri}"
-
-    @pytest.mark.parametrize(
-        "delegate_uris_public_keys",
-        [
-            (PUBLIC_DHT_DELEGATE_URIS, PUBLIC_DHT_PUBLIC_KEYS),
-        ],
-        indirect=True,
-    )
-    def test_communication_direct(self, delegate_uris_public_keys):
-        """Test communication direct (i.e. both clients registered to same peer)."""
-
-        for uri, peer_public_key in zip(*delegate_uris_public_keys):
-            connection1 = _make_libp2p_client_connection(
-                peer_public_key=peer_public_key, uri=uri
-            )
-            multiplexer1 = Multiplexer([connection1])
-            multiplexer1.connect()
-            self.multiplexers.append(multiplexer1)
-
-            connection2 = _make_libp2p_client_connection(
-                peer_public_key=peer_public_key, uri=uri
-            )
-            multiplexer2 = Multiplexer([connection2])
-            multiplexer2.connect()
-            self.multiplexers.append(multiplexer2)
-
-            sender = connection1.address
-            to = connection2.address
-            envelope = self.enveloped_default_message(to=to, sender=sender)
-
-            multiplexer1.put(envelope)
-            delivered_envelope = multiplexer2.get(block=True, timeout=20)
-            assert self.sent_is_delivered_envelope(envelope, delivered_envelope)
-
-    @pytest.mark.parametrize(
-        "delegate_uris_public_keys",
-        [
-            (PUBLIC_DHT_DELEGATE_URIS, PUBLIC_DHT_PUBLIC_KEYS),
-        ],
-        indirect=True,
-    )
-    def test_communication_indirect(self, delegate_uris_public_keys):
-        """Test communication indirect (i.e. clients registered to different peers)."""
-
-        delegate_uris, public_keys = delegate_uris_public_keys
-        assert len(delegate_uris) > 1, "Test requires at least 2 public dht node"
-
-        nodes = range(len(delegate_uris))
-        for i, j in itertools.permutations(nodes, 2):
-
-            connection1 = _make_libp2p_client_connection(
-                peer_public_key=public_keys[i],
-                uri=delegate_uris[i],
-            )
-            multiplexer1 = Multiplexer([connection1])
-            multiplexer1.connect()
-            self.multiplexers.append(multiplexer1)
-
-            sender = connection1.address
-
-            connection2 = _make_libp2p_client_connection(
-                peer_public_key=public_keys[j],
-                uri=delegate_uris[j],
-            )
-            multiplexer2 = Multiplexer([connection2])
-            multiplexer2.connect()
-            self.multiplexers.append(multiplexer2)
-
-            to = connection2.address
-            envelope = self.enveloped_default_message(to=to, sender=sender)
+    @classmethod
+    def setup_class(cls):  # overwrite the setup_class, reuse the tests
+        """Set up test"""
+        BaseP2PLibp2pTest.setup_class()
```

Comment:
because we inherit from `TestLibp2pConnectionPublicDHTRelay`, we do not call `super()` but instead the grandparent class `BaseP2PLibp2pTest` directly